### PR TITLE
ApartApp save apartment choice to backend

### PR DESCRIPTION
**Pull request by NerdyGriffin**
- [ ] Test this feature with API branch [apartapp-save-hall](https://github.com/gordon-cs/gordon-360-api/pull/452)
- [x] Comments and documentation

### CODE REVIEW

File: src/services/housing.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 
 import http from './http';
-import './user'; // Needed for typedef of StudentProfileInfo
+import './user';
```
Comment (NerdyGriffin) [inline src/services/housing.js:8]:
```suggestion
import './user'; // Needed for typedef of StudentProfileInfo
```
Add this comment back in, it was removed by mistake

File: src/services/housing.js [hunk 2]
```diff
@@ -78,23 +84,27 @@ const getApplicationID = async (username) => {
...
     Applicants: applicants.map((applicant) => applicant.Profile.AD_Username),
     // Applicants: applicants.map((applicant) => { Username: applicant.Profile.AD_Username, OffCampusProgram: applicant.OffCampusProgram }); // This is the correct code for when the backend has been updated expect the off-campus program info
-    ApartmentChoices: preferredHalls,
+    ApartmentChoices: apartmentChoices,
   };
-  return await http.post(`housing/apartment/save/`, applicationDetails);
+  if (applicationID === -1) {
+    return await http.post(`housing/apartment/save/`, applicationDetails);
+  } else {
+    return await http.put(`housing/apartment/save/`, applicationDetails);
```
Comment (Nick-Noormand) [inline src/services/housing.js:106]: I assume this is temporary since the if and else both do the same thing.

File: src/services/housing.js [hunk 3]
```diff
@@ -122,8 +132,8 @@ const getApartmentApplication = async (applicationID) => {
 
 export default {
   checkHousingStaff,
-  getApplicationID,
+  getCurrentApplicationID,
   saveApartmentApplication,
-  changeApplicationEditor,
+  changeApartmentAppEditor,
   getApartmentApplication,
 };
```
Comment (Nick-Noormand) [inline src/services/housing.js:139]: Besides that the naming convention changes and comments in this section look good.

File: src/views/ApartmentApp/components/StudentApplication/index.js [hunk 4]
```diff
@@ -626,18 +705,30 @@ const StudentApplication = ({ userProfile, authentication }) => {
...
-                      editorUsername={editorUsername}
-                      applicants={applicants}
-                      saving={saving}
-                      onSearchSubmit={handleSearchSubmit}
-                      onChangeEditor={handleChangeEditor}
-                      onApplicantRemove={handleApplicantRemove}
-                      onSaveButtonClick={handleSaveButtonClick}
-                      authentication={authentication}
-                    />
+                    {userProfile.AD_Username === editorUsername ? (
```
Comment (Nick-Noormand) [inline src/views/ApartmentApp/components/StudentApplication/index.js:708]: Does the user have to be the editor to see the full applicant list?

File: src/views/ApartmentApp/components/StudentApplication/index.js [hunk 5]
```diff
@@ -657,16 +748,26 @@ const StudentApplication = ({ userProfile, authentication }) => {
...
-                      authentication
-                      editorUsername={editorUsername}
-                      preferredHalls={preferredHalls}
-                      saving={saving}
-                      onHallAdd={handleHallAdd}
-                      onHallInputChange={handleHallInputChange}
-                      onHallRemove={handleHallRemove}
-                      onSaveButtonClick={handleSaveButtonClick}
-                    />
+                    {userProfile.AD_Username === editorUsername ? (
```
Comment (Nick-Noormand) [inline src/views/ApartmentApp/components/StudentApplication/index.js:751]: Okay so for non-editors we are making it that they can't see the application then. Got it.

**Review by Nick-Noormand — APPROVED** [review]
Just a couple of question I had in there, but overall looks good. Mostly just name changes and reformatting.

File: src/services/housing.js (hunk 2 above)
Comment (NerdyGriffin) [reply src/services/housing.js:106]: They do not do the same thing, one is "post" and the other is "put", which on the backend side directs them to different methods of the HousingController

**Pull request merged**